internal/pkg/cli/deploy/workload.go

```diff
@@ -369,16 +428,66 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 		return fmt.Errorf("login to image repository: %w", err)
 	}
 	out.ImageDigests = make(map[string]ContainerImageIdentifier, len(buildArgsPerContainer))
+
+	// create an array of output buffers, one for each container.
+	buffers := make([]*buildPushOutputBuffer, len(buildArgsPerContainer))
+
+	// create a context and an error group.
+	ctx := context.Background()
+	g, _ := errgroup.WithContext(ctx)
+
+	// counter for indexing the output buffers.
+	count := 0
+
+	// mutex for synchronizing access to the output map.
+	var mux sync.Mutex
```

Comment:
```suggestion
	var mu sync.Mutex
```
nit - and declare it next to the variable it protects (in this case, `out.ImageDigests` on L430)